### PR TITLE
add imagepull secret to kafka related pod

### PR DESCRIPTION
**Pull request by ldpliu**
https://issues.redhat.com/browse/ACM-9788

The operator image pull secret could be specified in catalogsource.

https://olm.operatorframework.io/docs/tasks/make-catalog-available-on-cluster/#using-registry-images-that-require-authentication-as-catalogbundleoperatoroperand-images

So there is no need to set it for kafka in globalhub part.

### CODE REVIEW

File: operator/pkg/transporter/strimzi_transporter.go [hunk 1]
```diff
@@ -905,6 +906,59 @@ func (k *strimziTransporter) setTolerations(mgh *operatorv1alpha4.MulticlusterGl
 	}
 }
 
+// setTolerations sets the kafka tolerations based on the mgh tolerations
```
Comment (hchenxa) [inline operator/pkg/transporter/strimzi_transporter.go:909]: try to update the comments here for setImagePullSecret

Comment (ldpliu) [reply operator/pkg/transporter/strimzi_transporter.go]: fixed

File: operator/pkg/transporter/strimzi_transporter.go [hunk 2]
```diff
@@ -905,6 +906,91 @@ func (k *strimziTransporter) setTolerations(mgh *operatorv1alpha4.MulticlusterGl
 	}
 }
 
+// setImagePullSecret sets the kafka image pull secret based on the mgh imagepullsecret
+func (k *strimziTransporter) setImagePullSecret(mgh *operatorv1alpha4.MulticlusterGlobalHub,
+	kafkaCluster *kafkav1beta2.Kafka,
+) {
+	if mgh.Spec.ImagePullSecret != "" {
```
Comment (clyang82) [inline operator/pkg/transporter/strimzi_transporter.go:913]: Can you use jsonpatch? for instance, https://github.com/stolostron/multicluster-global-hub/blob/15b8c2f58bbb01d10b2198723ca36aeeb4215bc0/operator/pkg/transporter/strimzi_transporter.go#L603

**Review by clyang82 — APPROVED** [review]
LGTM

**Pull request merged**